week_4/quiz-trivia-app/src/app/screens/quiz-page/answer/answer.component.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  AfterViewInit,
+  ChangeDetectorRef,
+  Component,
+  Input,
+  OnInit,
+} from '@angular/core';
+import { QuizService } from '../../../data/quiz.service';
+import { Quiz } from '../../../data/quiz.model';
+import { CommonModule } from '@angular/common';
+import { StorageService } from '../../../storage.service';
+
+@Component({
+  selector: 'app-answer',
+  standalone: true,
+  imports: [CommonModule],
+  templateUrl: './answer.component.html',
+  styleUrl: './answer.component.css',
+})
+export class AnswerComponent implements OnInit, AfterViewInit {
+  @Input() subjectObject!: Quiz | undefined;
+
+  num = 0;
+  answers!: string[];
+  currentAnswer = this.subjectObject?.questions[this.num].answer;
+  selectedAnswer: string | null = null;
+  letter: string[] = ['A', 'B', 'C', 'D'];
+  hoverIndex = -1;
+  selectIndex = -1;
+  isCorrect!: boolean;
+  isSubmitted = false;
+  renderWarning = false;
+  correctAnswer!: string;
+  correctAnswerIndex!: number;
+  totalQuestions: number | undefined = undefined;
+  score = 0;
+
+  constructor(
+    private quizService: QuizService,
+    private storageService: StorageService,
+    private cdr: ChangeDetectorRef
+  ) {}
+
+  ngOnInit(): void {
+    setTimeout(() => {
+      const storedCorrectAnswerIndex =
+        this.storageService.getData<number>('correctAnswerIndex');
+      this.correctAnswerIndex = storedCorrectAnswerIndex ?? -1;
+
+      const storedSubmitState =
+        this.storageService.getData<boolean>('isSubmitted');
+      this.isSubmitted = storedSubmitState ?? false;
+    });
+
+    const savedAnswers =
+      this.storageService.getData<string[]>('possibleAnswers');
+    this.answers = savedAnswers ?? [];
+
+    const storedSubject = this.storageService.getData<Quiz>('selectedSubject');
+    this.subjectObject = storedSubject ?? undefined;
+
+    const storedScore = this.storageService.getData<number>('scorePoints');
+    this.score = storedScore ?? 0;
+
+    const storedIndex = this.storageService.getData<number>(
+      'currentQuestionIndex'
+    );
+    this.num = storedIndex ?? 0;
+
+    const storedSubmitState =
+      this.storageService.getData<boolean>('isSubmitted');
+    this.isSubmitted = storedSubmitState ?? false;
+
+    const storedAnswer = this.storageService.getData<string>('selectedAnswer');
+    this.selectedAnswer = storedAnswer ?? null;
+
+    const storedIsCorrectState =
+      this.storageService.getData<boolean>('isCorrect');
+    this.isCorrect = storedIsCorrectState ?? false;
+
```

Comment:
with all variable above, assign value directly to local storage, they are redundant

Reply:
done